interfaces/apparmor/spec.go

```diff
@@ -56,6 +62,20 @@ type Specification struct {
 	// of the application or hook.
 	snippets map[string][]string
 
+	// prioritizedSnippets are just like snippets, but they have a priority value
+	// and an UID. An interface can add a snippet with a specific UID and a priority.
+	// If there doesn't exist any snippet with that UID, the passed snippet will be
+	// added as-is. But if it does exist, the new snippet will replace the old one if
+	// the new priority is bigger than the old one; will be appended if the new
+	// priority is the same as the old one, and will be discarded if the new priority
+	// is smaller than the old one.
+	prioritizedSnippets map[string]map[string]prioritizedSnippetsType
+
+	// SnippetKeys is a list of allowed keys for prioritized snippets.
+	// Trying to add a prioritized snippet with an unregistered key is
+	// an error. Trying to register the same key twice is also an error.
+	snippetKeys map[string]bool
```

Comment:
If you make snippet key a distinct type then the maps will be easier to read.

Reply:
Done.